### ISSUE TITLE
"bun update --latest" error

### ISSUE THREAD

**Issue by renzp94** [oven-sh/bun#11609]
### How can we reproduce the crash?

bun update --latest

### Relevant log output

```shell
bun update v1.1.12 (43f0913c)
============================================================
Bun v1.1.12 (43f0913c) macOS Silicon
Args: "bun" "update" "--latest"
Elapsed: 18ms | User: 19ms | Sys: 10ms
RSS: 20.58MB | Peak: 20.58MB | Commit: 1.07GB | Faults: 521

panic(main thread): Segmentation fault at address 0x347BC321BC8
oh no: Bun has crashed. This indicates a bug in Bun, not your code.

To send a redacted crash report to Bun's team,
please file a GitHub issue using the link below:

 https://bun.report/1.1.12/Mu143f0913AA__u/g3D+nh8DmyvB2lhB__A2u0Bxjy35jC

[1]    74186 trace trap  bun update --latest
```

### Stack Trace (bun.report)

Bun v1.1.12 ([`43f0913`](<https://github.com/oven-sh/bun/tree/43f0913c3875cbf260c60868268adc5953b09a80>)) on macos aarch64 [UpdateCommand]

Segmentation fault at address 0x3CC60372A08

- *2 unknown/js code*
- `src.install.install.PackageManager.updatePackageJSONAndInstallWithManager__anon_70529`
- `src.cli.update_command.UpdateCommand.exec`
- `src.cli.Cli.start`
- `main`
- *2 unknown/js code*

**Comment by dylan-conway** [comment]
Most likely fixed with #11547 . Can you try again with canary (`bun upgrade --canary`)?

**Comment by renzp94** [comment]
> Most likely fixed with #11547 . Can you try again with canary (`bun upgrade --canary`)?

Bun v1.1.13-canary.1 is Ok.Thanks.
